### ISSUE TITLE
`test_submit_timelimit` unit test for the local scheduler fails on Debian

### ISSUE THREAD

**Issue by vkarak** [eth-cscs/reframe#1494]
```
_________________________________________________________________________________________ test_submit_timelimit[local] _________________________________________________________________________________________

minimal_job = <reframe.core.schedulers.Job object at 0x7f4190ce6fd0>, local_only = None

    def test_submit_timelimit(minimal_job, local_only):
        minimal_job.time_limit = '2s'
        prepare_job(minimal_job, 'sleep 10')
        t_job = datetime.now()
        minimal_job.submit()
        assert minimal_job.jobid is not None
        minimal_job.wait()
        t_job = datetime.now() - t_job
>       assert t_job.total_seconds() >= 2
E       assert 1.971438 >= 2
E        +  where 1.971438 = <built-in method total_seconds of datetime.timedelta object at 0x7f4190d045a8>()
E        +    where <built-in method total_seconds of datetime.timedelta object at 0x7f4190d045a8> = datetime.timedelta(seconds=1, microseconds=971438).total_seconds

unittests/test_schedulers.py:302: AssertionError
=========================================================================================== short test summary info ============================================================================================
FAILED unittests/test_schedulers.py::test_submit_timelimit[local] - assert 1.971438 >= 2
```

The failure seems to be occasional.

**Comment by vkarak** [comment]
This might be fixed with #1402. I will test the PR against this bug as well.

**Comment by vkarak** [comment]
I confirm it is fixed as well. How to reproduce:

```
docker run -it debian:latest
```

```
apt update
apt install python3-pip
apt install git
git clone https://github.com/eth-cscs/reframe.git
cd reframe
./bootstrap.sh
for i in $(seq 100); do ./test_reframe.py unittests/test_schedulers.py::test_submit_timelimit || exit 1; done
```

Master fails after a couple of iteration, whereas #1402 does not fail at all.
